lib/utils.pm

```diff
@@ -264,6 +265,21 @@ sub get_netboot_mirror {
     return get_var('MIRROR_' . uc($m_protocol));
 }
 
+# Function that will add all test repos available and then logs all the patches needed
+sub add_test_repositories {
+
+    my @repos = @_;
+    for my $var (qw(OS_TEST_REPO SDK_TEST_REPO TCM_TEST_REPO WSM_TEST_REPO WE_TEST_REPO)) {
+        my $repo = get_var($var);
+        next unless $repo;
+        zypper_call("--no-gpg-check ar -f '$repo' test-repo-$var");
+        zypper_call("ref -r $repo");
+        zypper_call("patches -r $repo", log => "patches-$var.log");
+        zypper_call("se -t package -r $repo", exitcode => [0, 104], log => "packages-$var.log");
+    }
+}
+
+
```

Comment:
The `lib/utils.pm` file is getting quite big. If you don't mind you could move this function into its own "lib" module with a good name and proper documentation. Compare to `lib/registration.pm` for an example. But the change looks fine functionality-wise so if this is urgently needed a QAM colleague could just accept as is.